boards: nordic: nrf54l*_cpuapp/nrf54l15_cpuapp_ns/nrf9160: use common dts partitioning

nordicjm opened the pull request:
Adapts to common dts files for partition information

nordicjm committed:
dts: vendor: nordic: nrf91xx_partition: Adapt to sub-partitions

Updates to use sub-partitions for TF-M slots

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: nrf54l15: Change to use common dts partitioning file

Adds a common vendor dts file specifying the default partition
layout for nRF54L15-based cpuapp board targets and updates boards
to use this common file. This also drops the secure/non-secure split
in the partitioning as this was reducing NVM storage that is not
used by this board target and fixes the wrong flash field in
twister yaml files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: nrf54l15_ns: Change to use common dts partitioning file

Adds a common vendor dts file specifying the default partition
layout for nRF54L15-based cpuapp_ns board targets and updates boards
to use this common file.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

Vge0rge commented on dts/vendor/nordic/nrf54l15_partition.dtsi:
```diff
@@ -0,0 +1,35 @@
+			reg = <0xb2000 DT_SIZE_K(648)>;
+		};
+
+		/* 32KiB from 0x154000 to 0x15bfff reserved for TF-M partitions */
```
This comment now seems invalid since this is not the ns target. Maybe its bettter to remove the comment and use the 32KiB in the other partitions.

nordicjm committed:
dts: vendor: nordic: nrf54l10: Fix wrong size of NVM

Fixes the NVM size going beyond what the chip supports

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: nrf54l10: Change to use common dts partitioning file

Adds a common vendor dts file specifying the default partition
layout for nRF54L10-based cpuapp board targets and updates boards
to use this common file. This also drops the secure/non-secure split
in the partitioning as this was reducing NVM storage that is not
used by this board target and fixes the wrong flash field in
twister yaml files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: nrf54l05: Change to use common dts partitioning file

Adds a common vendor dts file specifying the default partition
layout for nRF54L05-based cpuapp board targets and updates boards
to use this common file. This also drops the secure/non-secure split
in the partitioning as this was reducing NVM storage that is not
used by this board target and fixes the wrong flash field in
twister yaml files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: nrf54l10_cpuapp_ns: Add workaround for partitions

Adds a workaround for partitions on nrf54l10 non-secure devices to
prevent build issues, this needs to be fixed properly when a TF-M
update is done which sets the partition sizes properly

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: fs: fs_sample: Fix nrf54l15 dts overlay files

Fixes these files to no lonegr delete non-secure partitions
as these no longer exist

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
